python/extensions/python.bzl

```diff
@@ -88,14 +129,39 @@ Please define a toolchain that has is_version set to True.""")
  module has a toolchain of the same name.""".format(toolchain_attr.name))
             continue
         toolchain_names.append(name)
-        _python_register_toolchains(toolchain_attr, True)
-
-    # Create the hub for the interpreters and the
-    # the default toolchain.
+        toolchains.append(
+            _python_register_toolchains(
+                toolchain_attr,
+                version_constraint = True,
+            ),
+        )
+        python_versions[toolchain_attr.python_version] = toolchain_attr.name
```

Comment:
Give the root module precedence over submodules. Similarly, given earlier modules precedence over later modules (modules are iterated in breadth-first order). So don't overwrite an entry in `python_versions`; warn instead.

I'm fine with putting a TODO for this for now.